### PR TITLE
Update feature availability for dns-pod-service

### PR DESCRIPTION
**Pull request by chenxuc**
Removed the outdated availability section and add feature state
shortcode.

This is for issue #28059

### CODE REVIEW

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-28131--kubernetes-io-master-staging.netlify.app/docs/concepts/services-networking/dns-pod-service/

/kind cleanup

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ content_type: concept
 weight: 20
 ---
 <!-- overview -->
+
+{{< feature-state for_k8s_version="v1.14" state="stable" >}}
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:11]: Hi @chenxuc .
What do you think about moving this feature-state shortcode to the 
`Pod's DNS Config` section (around line 267):

Comment (chenxuc) [reply content/en/docs/concepts/services-networking/dns-pod-service.md]: Thanks for the review. I have moved the shortcode to the Pod's DNS Config section.

**Comment by kbhawkey** [comment]
Thanks @chenxuc .
/lgtm
/approve

**Pull request merged**